packages/map-template/src/components/Map/MapboxMap/ViewmodeSwitch/ViewModeSwitch.jsx

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useState } from 'react';
+import { useRecoilState, useRecoilValue } from 'recoil';
+import solutionState from '../../../../atoms/solutionState';
+import { ReactComponent as Light2D } from '../../../../assets/2d-light.svg';
+import { ReactComponent as Dark2D } from '../../../../assets/2d-dark.svg';
+import { ReactComponent as Light3D } from '../../../../assets/3d-light.svg';
+import { ReactComponent as Dark3D } from '../../../../assets/3d-dark.svg';
+import primaryColorState from '../../../../atoms/primaryColorState';
+import mapboxViewModeState from '../../../../atoms/mapboxViewModeState';
+import './ViewModeSwitch.scss';
+import currentPitchSelector from '../../../../selectors/currentPitch';
+
+export const ViewModes = Object.freeze({
+    'initial3D': 1,
+    'clicked3D': 2,
+    'clicked2D': 3
+});
+
+function ViewModeSwitch({ mapView }) {
+
+    const solution = useRecoilValue(solutionState);
+    const primaryColor = useRecoilValue(primaryColorState);
+    const [viewMode, setViewMode] = useRecoilState(mapboxViewModeState);
+    const [shouldShowSwitch, setShouldShowSwitch] = useState(false);
+    const currentPitch = useRecoilValue(currentPitchSelector);
+
+    useEffect(() => {
+        if (['mapbox', '3dwalls', 'floorplan'].every(requiredModule => solution.modules.map(module => module.toLowerCase()).includes(requiredModule))) {
+            setShouldShowSwitch(true);
+            setViewMode(ViewModes.initial3D);
+        }
+    }, [solution]);
+
+    useEffect(() => {
+        // Toggle 2D and 3D features on the map
+        if (mapView?.isReady) {
+            // If the viewMode is null, set the ViewMode to be initial3D
+            if (viewMode === null) {
+                setViewMode(ViewModes.initial3D);
```

Comment:
Could we have this as the default atom value? 🤔

Reply:
Yes, refactored the code. ✅